### PR TITLE
[authn_mappings api] update documentation for recent changes [DO NOT MERGE UNTIL 03/31]

### PR DESCRIPTION
**Pull request by nspringdd**
### What does this PR do?

This PR updates the authn_mapping API documentation to match recent changes. The changes are listed as follows:

```
renaming meta page total → total_count

addition of total_filtered_count

additional sort methods: roles.name, saml_assertion_attributes.attribute_key, saml_assertion_attributes.attribute_value

removal of role_uuid from responses

removal of role_uuid from requests

receiving role uuid from relationship field in requests for POST and pATCH
```

This PR also does some minor corrections (modified_at, and optional fields
, and no default when applicable).
### Preview link

https://docs-staging.datadoghq.com/nathan/authn-mapping-doc-updates/account_management/authn_mapping/

### CODE REVIEW

File: content/en/account_management/authn_mapping/_index.md [hunk 1]
```diff
@@ -222,7 +231,14 @@ Replace the `<YOUR_DATADOG_API_KEY>` and `<YOUR_DATADOG_APPLICATION_KEY>` placeh
       }
     }
   ],
-  "meta": {"page": {"total": 0, "page_number": 0, "page_size": 0}}
+  "meta": {
+    "page": {
+      "total_count": 1, 
+      "total_filtered_count": 1, 
+      "page_number": 0, 
+      "page_size": 0
```
Comment (pietdaniel) [inline content/en/account_management/authn_mapping/_index.md]: is this consistent with the users + roles API?

Comment (nspringdd) [reply content/en/account_management/authn_mapping/_index.md]: Users and roles API do not have a page_number and page_size, the other two fields are correct. I can remove them if they're causing confusion. AFAIK they're not being used by the frontend.

File: content/en/account_management/authn_mapping/_index.md [hunk 2]
```diff
@@ -328,13 +342,13 @@ Updates the AuthN Mapping `role`, `saml_assertion_attribute_id`, or both from a
...
   Replace `{UUID}` with the UUID of the AuthN Mapping you want to update.
-* **`id`** [*required*, *default*=none]:
+* **`id`** [*required*, no default]:
   The UUID of the AuthN Mapping being updated. This property must match the `{UUID}` path parameter in the request.
-* **`role_uuid`** [*required*, *default*=none]:
-  The Roles API can be used to create and manage Datadog roles, what global permissions they grant, and which users belong to them. When you create a Role, it is assigned a UUID. For more information about finding the `role_uuid` for the role you are updating, see the [Role API documentation][1].
-* **`attribute_key`** [*required*, *default*=none]:
+* **`roles.uuid`** [*optional*, *default*=none]:
+  The `UUID` of the Role to map to. The Roles API can be used to create and manage Datadog roles, what global permissions they grant, and which users belong to them. Note that this attribute should be presented as part of a `roles` relationship block in requests. See the example below for more details. When you create a Role, it is assigned a UUID. For more information about finding the `UUID` for the role you want to map to, see the [Role API documentation][1].
```
Comment (apigirl) [inline content/en/account_management/authn_mapping/_index.md]:
```suggestion
  The `UUID` of the Role to map to. The Roles API can be used to create and manage Datadog roles, what global permissions they grant, and which users belong to them. 
 **Note**: This attribute should be presented as part of a `roles` relationship block in requests. See the example below for more details. When you create a Role, it is assigned a UUID. For more information about finding the `UUID` for the role you want to map to, see the [Role API documentation][1].
```

File: content/en/account_management/authn_mapping/_index.md [hunk 3]
```diff
@@ -320,43 +342,49 @@ Replace the `<YOUR_DATADOG_API_KEY>` and `<YOUR_DATADOG_APPLICATION_KEY>` placeh
...
-* **`role_uuid`** [*required*, *default*=none]:
-  The Roles API can be used to create and manage Datadog roles, what global permissions they grant, and which users belong to them. When you create a Role, it is assigned a UUID. For more information about finding the `role_uuid` for the role you are updating, see the [Role API documentation][1].
-* **`attribute_key`** [*required*, *default*=none]:
+* **`{authn_mapping_id}`** [*required*, no default]:
+  Replace `{authn_mapping_id}` with the ID of the AuthN Mapping you want to update. This is required in both the path of the request and the body of the request.
+* **`role["data"]["id"]`** [*optional*, *default*=none]:
+ The `ID` of the Role to map to. The Roles API can be used to create and manage Datadog roles, what global permissions they grant, and which users belong to them. 
+ **Note**: This attribute should be presented as part of a `role` relationship block in requests. See the example below for more details. When you create a Role, it is assigned a ID. For more information about finding the `ID` for the role you want to map to, see the [Role API documentation][1].
```
Comment (cswatt) [inline content/en/account_management/authn_mapping/_index.md]:
```suggestion
 **Note**: This attribute should be presented as part of a `role` relationship block in requests. See the example below for more details. When you create a Role, it is assigned an ID. For more information about finding the `ID` for the role you want to map to, see the [Role API documentation][1].
```

File: content/en/account_management/authn_mapping/_index.md [hunk 4]
```diff
@@ -459,6 +493,109 @@ HTTP/2 204
...
+| Method   | Endpoint path               | Required payload |
+|----------|-----------------------------|------------------|
+| `POST`   | `/v1/org_preferences`       | JSON             |
+
+##### ARGUMENTS
+
+* **`{preference_type}`** [*required*, no default]:
+  Preference to update, required to be "saml_authn_mapping_roles"
+* **`{preference_data}`** [*required*, no default]:
+  Data to update preference with, must be true or false. true to enable all mappings, false to disable
```
Comment (cswatt) [inline content/en/account_management/authn_mapping/_index.md]:
```suggestion
  Data to update preference with, must be true or false: true to enable all mappings, false to disable
```

**Pull request merged**